tests/app/views/test_frameworks.py

```diff
@@ -586,6 +579,50 @@ def test_list_suppliers_with_agreements_returned(self):
             times = [parse_time(item['agreementReturnedAt']) for item in data['supplierFrameworks']]
             assert times[0] > times[1]
 
+    def test_list_suppliers_with_agreements_returned_uses_framework_agreement_timestamp(self):
+        with self.app.app_context():
+            # Set supplier 3 agreement_returned_at as the highest time of suppliers 3 and 4 so we can test that
+            # FrameworkAgreement.signed_agreement_returned_at takes priority over
+            # SupplierFramework.agreement_returned_at when ordering by time
+            db.session.execute(
+                "UPDATE supplier_frameworks SET agreement_returned_at='{}' WHERE supplier_id=3".format(
+                    datetime.datetime.utcnow())
+            )
+            db.session.commit()
+
+            response = self.client.get('/frameworks/g-cloud-7/suppliers?agreement_returned=true')
+
+            assert response.status_code == 200
+            data = json.loads(response.get_data())
+            assert len(data['supplierFrameworks']) == 2
+
+            times = [parse_time(item['agreementReturnedAt']) for item in data['supplierFrameworks']]
+            assert times[0] > times[1]
+
+            # Check SupplierFramework.agreement_returned_at is not used (indicated by supplier 4
+            # appearing before supplier 3 in our results)
+            supplier_ids = [item['supplierId'] for item in data['supplierFrameworks']]
+            assert supplier_ids[0] > supplier_ids[1]
+
+    def test_list_suppliers_with_agreements_returned_includes_those_with_only_supplier_framework(self):
+        with self.app.app_context():
+            # Set supplier 2 to have their agreement returned information only set in the SupplierFramework table
+            db.session.execute(
```

Comment:
How do we know there's nothing in the FrameworkAgreement table for this supplier? It doesn't seem obvious to me from the test setup here. Does our also need to execute a delete from the FrameworkAgreement table to be sure there are no entries there?